crates/matrix-sdk/src/widget/matrix.rs

```diff
@@ -76,17 +83,30 @@ impl MatrixDriver {
         });
 
         let messages = self.room.messages(options).await?;
-        Ok(messages.chunk.into_iter().map(|ev| ev.into_raw().cast()).collect())
+        Ok(messages
+            .chunk
+            .into_iter()
+            .map(|ev| ev.into_raw().cast())
+            .filter(|ev| match &state_key {
```

Comment:
Any chance this can be fused into a `filter_map`?

Reply:
I could, though what would be the benefit? My understanding is that operations on Rust iterators are fused automatically.